milatools/cli/code.py

```diff
@@ -0,0 +1,195 @@
+    if job or node:
+        if job and node:
+            logger.warning(
+                "Both job ID and node name were specified. Ignoring the node name and "
+                "only using the job id."
+            )
+        job_id_or_node = job or node
+        assert job_id_or_node is not None
+        compute_node_task = ComputeNode.connect(
+            login_node=login_node, job_id_or_node_name=job_id_or_node
+        )
+    else:
+        if cluster in DRAC_CLUSTERS and not any("--account" in flag for flag in alloc):
+            logger.warning(
+                "Warning: When using the DRAC clusters, you usually need to "
+                "specify the account to use when submitting a job. You can specify "
+                "this in the job resources with `--alloc`, like so: "
+                "`--alloc --account=<account_to_use>`, for example:\n"
+                f"mila code some_path --cluster {cluster} --alloc "
+                f"--account=your-account-here"
+            )
+        # Set the job name to `mila-code`. This should not be changed by the user
+        # ideally, so we can collect some simple stats about the use of `milatools` on
+        # the clusters.
+        if any(flag == "-J" or "-J=" in flag or "--job-name" in flag for flag in alloc):
+            # todo: Get the job name from the flags instead?
+            raise MilatoolsUserError(
+                "The job name flag (--job-name or -J) should be left unset for now "
+                "because we use the job name to measure how many people use `mila "
+                "code` on the various clusters. We also make use of the job name when "
+                "the call to `salloc` is interrupted before we have a chance to know "
+                "the job id."
+            )
+        job_name = "mila-code"
+        alloc = alloc + [f"--job-name={job_name}"]
+
+        if persist:
+            compute_node_task = sbatch(
+                login_node, sbatch_flags=alloc, job_name=job_name
+            )
+        else:
+            # NOTE: Here we actually need the job name to be known, so that we can
+            # scancel jobs if the call is interrupted.
+            compute_node_task = salloc(
+                login_node, salloc_flags=alloc, job_name=job_name
+            )
+
+    if sync_vscode_extensions_task is not None:
+        # Sync the vscode extensions at the same time as waiting for the job.
+        # Wait until all extensions are done syncing before launching vscode.
+        # If any of the tasks failed, we want to raise the exception.
+        # NOTE: Not using this at the moment because when interrupted, the job request
+        # isn't cancelled properly.
+        compute_node, _ = await asyncio.gather(
+            compute_node_task,
+            sync_vscode_extensions_task,
+        )
+    else:
+        compute_node = await compute_node_task
+
+    await launch_vscode_loop(code_command, compute_node, path)
```

Comment:
KeyboardInterrupt and asyncio.CancelledError are caught inside the loop, and KeyboardInterrupt is also being caught at the top-level in `commands.py`. I'd be wary of adding other try-excepts, I'm honestly already having some trouble understanding precisely how interrupts are handled by asyncio. 

Instead of adding a new try-except here, in 414b3de35dcd9c70b5ca6e107d055543c3e2a24f I broadened the try-except in the vscode loop, so that it will return if there is an unexpected error, and we will print the persist instructions.